main/qpdf/APKBUILD

```diff
@@ -1,15 +1,14 @@
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 pkgname=qpdf
-pkgver=8.3.0
+pkgver=8.4.0
 pkgrel=0
 pkgdesc="Command-line tools and library for transforming PDF files"
 url="http://qpdf.sourceforge.net"
 arch="all"
 license="Apache-2.0"
-depends=""
+depends="qpdf-libs=$pkgver-r$pkgrel"
```

Comment:
Why did you add this?

Reply:
@clandmeter my reasoning is:

1. qpdf depends upon qpdf-libs, for example see [pkgs.a.o](https://pkgs.alpinelinux.org/package/edge/main/aarch64/qpdf)
2. if the version of qpdf-libs does not exactly match the version of qpdf, qpdf will fail to run see detailed message below
3. this is the best way I know of to make sure the versions match

<details>

```
$ qpdf --verson
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper21getFormXObjectForPageEb: symbol not found
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper16placeFormXObjectE16QPDFObjectHandleNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS0_9RectangleEb: symbol not found
Error relocating /usr/bin/qpdf: _ZN10QPDFWriter25setR6EncryptionParametersEPKcS1_bbbbbb15qpdf_r3_print_eb: symbol not found
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper12getAttributeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEb: symbol not found
Error relocating /usr/bin/qpdf: _ZNK4QPDF11getUniqueIdEv: symbol not found
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper15shallowCopyPageEv: symbol not found
Error relocating /usr/bin/qpdf: _ZN10QPDFWriter25setR4EncryptionParametersEPKcS1_bbbbbb15qpdf_r3_print_ebb: symbol not found
Error relocating /usr/bin/qpdf: _ZN10QPDFWriter25setR5EncryptionParametersEPKcS1_bbbbbb15qpdf_r3_print_eb: symbol not found
Error relocating /usr/bin/qpdf: _ZN16QPDFObjectHandle21getUniqueResourceNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERi: symbol not found
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper23externalizeInlineImagesEm: symbol not found
Error relocating /usr/bin/qpdf: _ZN20QPDFPageObjectHelper10getTrimBoxEb: symbol not found
Error relocating /usr/bin/qpdf: _ZN5QUtil27possible_repaired_encodingsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE: symbol not found
Error relocating /usr/bin/qpdf: _ZN10QPDFWriter25setR3EncryptionParametersEPKcS1_bbbbbb15qpdf_r3_print_e: symbol not found
Error relocating /usr/bin/qpdf: _ZN5QUtil15utf8_to_pdf_docERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERS5_c: symbol not found
Error relocating /usr/bin/qpdf: _ZN5QUtil16analyze_encodingERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERbS8_S8_: symbol not found
```
</details>

From my brief investigation I think the reason for the error is that qpdf dynamically links modules with dlopen which abuild doesn't detect. I think that without the change to depends above abuild / apk do not require a match in the versions which can lead to errors when more than one version is available / installed.

Happy to be corrected if I have misunderstood anything. Thanks for reviewing.

Reply:
I cannot reproduce your issue, there is probably something wrong with your setup.

```
scanelf --needed `which qpdf`
 TYPE   NEEDED FILE
ET_DYN libqpdf.so.21,libstdc++.so.6,libgcc_s.so.1,libc.musl-x86_64.so.1 /usr/bin/qpdf
```

It looks like it links just fine, so you can just remove that part.

Reply:
Thanks for explaining so clearly on IRC; I made the change and CI still passes 😄